### DISCUSSION TITLE
Using `var` when `override`ing

### DISCUSSION THREAD

**Discussion by TonyValenti** [dotnet/csharplang#4702] [Language Ideas]
Today, if you change the return type on a virtual method, you have to find/replace everywhere that method is overridden with the new type.  It would be nice if ```var``` could be used on overrides to mean "Use the same type as the base type".

For example:

```
public class C0 {
  public virtual int M(){
    return 0;
  }
}

public class C1 : C0 {
  public override var M(){ //var = int
    return 1;
  }
}
```

In the above code, I can go to C0 and change ```int``` to ```long``` and C1's M picks up that change.

**Comment by jnm2** [comment]
I believe GitHub titles render correctly when you use the normal inline code markdown (surrounding with single `` ` ``).

**Reply by jnm2** [reply]
Sorry about that, looks like it only works in PRs and issue titles.
![image](https://user-images.githubusercontent.com/8040367/116294118-964b8800-a765-11eb-8315-16c8270af24c.png)

**Comment by jnm2** [comment]
This is interesting. Would `var` gain the ability to mean `void`?

**Reply by TonyValenti** [reply]
yes. It would be whatever the the type of the parent method is.

**Comment by HaloFour** [comment]
IMO it sounds like something better solved by improvements to the refactoring tools when it comes to virtual/overridden members.  The Change Signature refactoring in IntelliJ already handles this in Java, I'd be surprised if R# didn't do the same for C#.
